timeseries/src/autogluon/timeseries/metrics/point.py

```diff
@@ -246,8 +246,11 @@ def compute_metric(
         if self._past_abs_seasonal_error is None:
             raise AssertionError("Call `save_past_metrics` before `compute_metric`")
 
-        mae_per_item = (y_true - y_pred).abs().groupby(level=ITEMID, sort=False).mean()
-        return self._safemean(mae_per_item / self._past_abs_seasonal_error)
+        num_items = len(self._past_abs_seasonal_error)
+        # Reshape MAE values into [num_items, prediction_length] to normalize per item without groupby
+        mae_reshaped = np.abs(y_true.values - y_pred.values).reshape([num_items, -1])
+        # We assume that items are in the same order in both arrays because predictor sorts by item_id
+        return self._safemean(mae_reshaped / self._past_abs_seasonal_error.values[:, None])
```

Comment:
This change is necessary to make sure that our MASE/SQL implementation aligns with GluonTS if missing values are present. For example, suppose that `prediction_length=100` and we have two items: 
- item A with 1 observed value in the forecast horizon
- item B with 100 observed values in the forecast horizon

In our old implementation with groupby, we'd first compute the mean for each item, and then take mean across items. This would assign 100x more weight to the single value for item A compared to the values in item B.

In the new implementation, each observed value in the forecast horizon is weighted equally (same as in GluonTS).